### ISSUE TITLE
Extract some public APIs from torch::cuda::initModule(module) to torch::initModule()

### ISSUE THREAD

**Issue by Danielmic** [pytorch/pytorch#129027]
### 🚀 The feature, motivation and pitch

Currently there are some APIs in torch::cuda::initModule(module) that are device-independent, such as the following:
`
static void registerCudaPluggableAllocator(PyObject* module) {
  auto m = py::handle(module).cast<py::module>();  
  //  ...
  m.def("_set_cached_tensors_enabled", [](bool enabled) {
    at::caching::set_cached_tensors_enabled(enabled);
  });

  m.def("_add_cached_tensor", [](const at::Tensor& t) {
    at::caching::add_cached_tensor(t);
  });

  m.def("_remove_cached_tensor", [](const at::Tensor& t) {
    at::caching::remove_cached_tensor(t);
  });

  m.def("_is_cached_tensor", [](const at::Tensor& t) {
    return at::caching::is_cached_tensor(t);
  });
  // ...
}
`

### Alternatives

Can we extract these APIs so that they can be used by all devices?

**Comment by albanD** [comment]
@guangyey are these already exposed to the generic autocast API?

**Comment by guangyey** [comment]
@albanD  No. These API functionalities are generic but defined in `torch/csrc/cuda/Module.cpp` and only exposed when `USE_CUDA=ON`. Decoupling them with CUDA building is better.

**Comment by guangyey** [comment]
I draft a PR here https://github.com/pytorch/pytorch/pull/129357
